extensions/nginx/index.js

```diff
@@ -131,28 +131,48 @@ class NginxExtension extends cli.Extension {
+                const acmeTmpDir = path.join(os.tmpdir(), 'acme.sh');
+
+                this.ui.logVerbose('ssl: cloning acme.sh to temporary directory', 'green');
+                return execa.shell(`git clone https://github.com/Neilpang/acme.sh.git ${acmeTmpDir}`).then(() => {
+                    this.ui.logVerbose('ssl: installing acme.sh components');
+
+                    // Installs acme.sh into /etc/letsencrypt
+                    return this.ui.sudo('./acme.sh --install --home /etc/letsencrypt', {cwd: acmeTmpDir});
+                }).catch((error) => Promise.reject(new cli.errors.ProcessError(error)));
+            }
+        }, {
             title: 'Getting SSL Certificate from Let\'s Encrypt',
             skip: (ctx) => ctx.dnsfail,
             task: () => {
-                return execa.shell('curl https://get.acme.sh | sh').then(() => {
-                    const acmeScriptPath = path.join(os.homedir(), '.acme.sh', 'acme.sh');
+                const cmd = `/etc/letsencrypt/acme.sh --issue --home /etc/letsencrypt --domain ${parsedUrl.hostname} --webroot ${rootPath} ` +
+                `--reloadcmd "nginx -s reload" --accountemail ${argv.sslemail}${argv.sslstaging ? ' --staging' : ''}`;
 
-                    const cmd = `${acmeScriptPath} --issue --domain ${parsedUrl.hostname} --webroot ${rootPath} ` +
-                        `--accountemail ${argv.sslemail}${argv.sslstaging ? ' --staging' : ''}`;
+                return this.ui.sudo(cmd).catch((error) => {
+                    if (error.code === 2) {
+                        // error code 2 is given if a cert doesn't need to be renewed
+                        return Promise.resolve();
+                    }
 
-                    return execa.shell(cmd);
-                }).catch((error) => {
-                    // Certs have been generated before, skip
-                    if (!error.stdout.match(/Skip/)) {
-                        return Promise.reject(new cli.errors.ProcessError(error));
+                    if (error.stderr.match(/Verify error:Invalid Response/)) {
+                        // Domain verification failed
+                        return Promise.reject(new cli.errors.SystemError(
+                            'Your domain name is not pointing to the correct IP address of your server, please update it and run `ghost setup ssl` again'
+                        ));
                     }
+
+                    // It's not an error we expect might happen, throw a ProcessError instead.
+                    return Promise.reject(new cli.errors.ProcessError(error));
                 });
             }
         }, {
             title: 'Generating Encryption Key (may take a few minutes)',
             skip: (ctx) => ctx.dnsfail || fs.existsSync(dhparamFile),
             task: () => {
-                return this.ui.sudo(`openssl dhparam -out ${dhparamFile} 2048 > /dev/null`)
+                return this.ui.sudo(`openssl dhparam -out ${dhparamFile} 2048`)
```

Comment:
This change looks like it doesn't belong here?

Reply:
yeah - i'll fix it in a separate PR. That redirect actually does nothing at the moment 😁